internal/rename/controller.go

```diff
@@ -0,0 +1,59 @@
+/*
+ *  *******************************************************************************
+ *  * Copyright (c) 2019 Edgeworx, Inc.
+ *  *
+ *  * This program and the accompanying materials are made available under the
+ *  * terms of the Eclipse Public License v. 2.0 which is available at
+ *  * http://www.eclipse.org/legal/epl-2.0
+ *  *
+ *  * SPDX-License-Identifier: EPL-2.0
+ *  *******************************************************************************
+ *
+ */
+
+package rename
+
+import (
+	"fmt"
+	apps "github.com/eclipse-iofog/iofog-go-sdk/pkg/apps"
+	"github.com/eclipse-iofog/iofog-go-sdk/pkg/client"
+	"github.com/eclipse-iofog/iofogctl/internal"
+	"github.com/eclipse-iofog/iofogctl/internal/config"
+	"github.com/eclipse-iofog/iofogctl/pkg/util"
+)
+
+type controllerExecutor struct {
+	namespace string
+	name      string
+	newname   string
+}
+
+func newControllerExecutor(namespace, name, newname string) *controllerExecutor {
+	c := &controllerExecutor{}
+	c.namespace = namespace
+	c.name = name
+	c.newname = newname
+	return c
+}
+
+func (exe *controllerExecutor) GetName() string {
+	return exe.name
+}
+
+func (exe *controllerExecutor) Execute() error {
+	// Check that Controller exists in current namespace
+	_, err := config.GetController(exe.namespace, exe.name)
+	if err != nil {
+		return err
+	}
+
+	util.SpinStart(fmt.Sprintf("Renaming Controller %s", name))
+
+	// Do a shallow rename of controller
+	err = RenameController(exe.namespace, exe.name, exe.newname)
```

Comment:
Same as above.

Reply:
In utils.go, I can move it into the executor function